packages/node/src/utils/project.ts

```diff
@@ -112,3 +118,26 @@ export function loadChainTypesFromJs(
   }
   return rawContent;
 }
+
+function dsContainsNonEventHandlers(ds: SubstrateProjectDs): boolean {
+  if (isRuntimeDs(ds)) {
+    return !!ds.mapping.handlers.find(
+      (handler) => handler.kind !== SubstrateHandlerKind.Event,
+    );
+  } else if (isCustomDs(ds)) {
+    // TODO this can be improved upon in the future.
+    return true;
+  }
+  return true;
+}
+
+export function isOnlyEventHandlers(project: SubqueryProject): boolean {
```

Comment:
We could define event only handler in this way, but we could face another issue, we could not know whether user try to access block content from event which we have wrapped previously. 

const wrappedEvents = wrapEvents(wrappedExtrinsics, events, wrappedBlock);

We can test this scenarios with starter, and we found error was thrown, but not clearly enough, could be very difficult to debug.

https://github.com/subquery/subql-starter/blob/96b09b6aa3a0c6506c423acf628f117cd47e9f83/Polkadot/Polkadot-starter/src/mappings/mappingHandlers.ts#L41

<img width="1467" alt="image" src="https://github.com/subquery/subql/assets/10431657/a1f76bdd-b101-4703-84c5-72f1cebca7c0">

I think we could implement within wrapped event, if block content is accessed in handler, we could provide more friendly warning/error. Or even refetch the block content.

Reply:
This is why there is the `skipBlock` option, users have to opt into it and test it themselves. They should update their handler types accordingly